src/init.cpp

```diff
@@ -902,6 +902,12 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
             return InitError(strprintf(_("Invalid amount for -minrelaytxfee=<amount>: '%s'"), mapArgs["-minrelaytxfee"]));
     }
 
+    bool fStandard = !GetBoolArg("-acceptnonstdtxn", !Params().RequireStandard());
+    // If we specified an override but that override was not accepted then its an error
+    if (fStandard != Params().RequireStandard())
```

Comment:
This doesn't work in the case of regtest, where `-acceptnonstdtxn=false` should be accepted.  Use `if (Params().RequireStandard() && !fStandard)` here?

Reply:
Its accepted.  I tried it, but also here's how the logic works:

Params().RequireStandard() 
=> fRequireStandard || !GetBoolArg("-acceptnonstdtxn", true); (def of fn)
=> !GetBoolArg("-acceptnonstdtxn", true); (fRequireStandard == false)
=> true (because acceptnonstdtxn=false was an arg)

bool fStandard = !GetBoolArg("-acceptnonstdtxn", !Params().RequireStandard());
= true (because acceptnonstdtxn=false was an arg)

if (fStandard != Params().RequireStandard()) =>
if (true != true)

Reply:
Ok, it works because the parameter being validated is already affecting the behavior of CChainParams::RequireStandard().  I guess making CChainParams:fRequireStandard public would be worse.